File: week2/q7.py
def multiply(A, B):
    m = len(A)
    n = len(B)

    prod = [0] * (m + n - 1)

    for i in range(m):
        for j in range(n):
            prod[i + j] += A[i] * B[j]

    return prod

File: week2/q7.c
#include <stdio.h>
#include <stdlib.h>

int* multiply(int* A, int m, int* B, int n, int* resultSize) {
    int size = m + n - 1;
    int* prod = (int*) calloc(size, sizeof(int)); 

    for (int i = 0; i < m; i++) {
        for (int j = 0; j < n; j++) {
            prod[i + j] += A[i] * B[j];
        }
    }

    *resultSize = size;
    return prod; 
}
